polaris/hub/polarisfs.py

```diff
@@ -89,14 +101,21 @@ def ls(
 
         ls_path = self.sep.join([self.base_path, "ls", path])
 
+        if self.use_listings_cache and not self.refresh:
+            cached_listings = self.dircache.get(path, None)
+            if cached_listings is not None:
+                return cached_listings if detail else [d["name"] for d in cached_listings]
+
+            cached_listings = self.dircache.get(self._parent(path), None)
+            if cached_listings is not None:
+                self.dircache[path] = []
+                return cached_listings if detail else [d["name"] for d in cached_listings]
```

Comment:
There is the [`_ls_from_cache`](https://github.com/fsspec/filesystem_spec/blob/518984d494e48f007d8aa48de882ed7014da2574/fsspec/spec.py#L354-L377) method in the parent class.